### PR TITLE
add support for UVT_DROP_PRIVILEGES/hvt_drop_privileges

### PR DESCRIPTION
**Pull request by adamsteen**

### CODE REVIEW

**Comment by mato** [comment]
@adamsteen Thanks. Can you try pushing something trivial to this branch to see if the CI gets unstuck? Not sure what happened there, have restarted things.

**Comment by mato** [comment]
@adamsteen Please push one more trivial change (or drop the one you made and force-push), CI is back up.

File: tenders/hvt/hvt_freebsd.c [hunk 1]
```diff
@@ -130,3 +130,8 @@ struct hvt *hvt_init(size_t mem_size)
     hvt->mem_size = mem_size;
     return hvt;
 }
+
+#if(HVT_DROP_PRIVILEGES == 1)
+void hvt_drop_privileges() {
```
Comment (mato) [inline tenders/hvt/hvt_freebsd.c]: Newline before `{`.

File: tenders/hvt/hvt_kvm.c [hunk 2]
```diff
@@ -93,3 +93,8 @@ struct hvt *hvt_init(size_t mem_size)
     hvt->b = hvb;
     return hvt;
 }
+
+#if(HVT_DROP_PRIVILEGES == 1)
+void hvt_drop_privileges() {
```
Comment (mato) [inline tenders/hvt/hvt_kvm.c]: Newline before `{`.

File: tenders/hvt/hvt_main.c [hunk 3]
```diff
@@ -196,5 +196,11 @@ int main(int argc, char **argv)
 
     setup_modules(hvt);
 
+#if(HVT_DROP_PRIVILEGES == 1)
+    hvt_drop_privileges();
+#else
+    warnx("Stern Warning, not recommended for production");
```
Comment (mato) [inline tenders/hvt/hvt_main.c]: `WARNING: Tender is configured with HVT_DROP_PRIVILEGES=0. Not dropping any privileges.`
`WARNING: This is not recommended for production use.`

File: tenders/hvt/hvt.h [hunk 4]
```diff
@@ -36,6 +36,10 @@
 #include "hvt_abi.h"
 #include "hvt_gdb.h"
 
+#ifndef HVT_DROP_PRIVILEGES
```
[the remark this answers is not in the document]
Comment (adamsteen) [reply tenders/hvt/hvt.h]: I have added a comment here, sorry English is not my strong suit.

File: tenders/hvt/hvt.h [hunk 5]
```diff
@@ -95,6 +99,13 @@ void hvt_mem_size(size_t *mem_size);
 void hvt_vcpu_init(struct hvt *hvt, hvt_gpa_t gpa_ep,
         hvt_gpa_t gpa_kend, char **cmdline);
 
+#if(HVT_DROP_PRIVILEGES == 1)
+/*
+ * Drop Privileges
+ */
```
[the remark this answers is not in the document]
Comment (adamsteen) [reply tenders/hvt/hvt.h]: Comment added.

File: tenders/hvt/hvt.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (mato) [reply tenders/hvt/hvt.h]: No worries, I can clean up the text later. Thanks.

**Pull request merged**